src/collection/components/CollectionOrAssignmentBlocks.tsx

```diff
@@ -0,0 +1,77 @@
+import { Container } from '@viaa/avo2-components';
+import { AssignmentBlock } from '@viaa/avo2-types/types/assignment';
+import { CollectionFragment } from '@viaa/avo2-types/types/collection';
+import React, { FC, ReactNode } from 'react';
+
+import { IconBar } from '../../shared/components';
+import { COLLECTION_OR_ASSIGNMENT_BLOCK_ICONS, CollectionBlockType } from '../collection.const';
+
+import { CollectionFragmentTypeItem, CollectionFragmentTypeText } from './index';
+
+interface CollectionOrFragmentBlocksProps {
+	blocks: CollectionFragment[] | AssignmentBlock[];
+	canPlay: boolean;
+	enableContentLinks: boolean;
```

Comment:
Vermijd het exposen van 1 variabele van gerenderde sub-componenten, gebruik een config object zoals:.
```
	config: { textItem: Partial<CollectionFragmentTypeTextProps>, ... }
```

Conceptueel zou ik persoonlijk nooit dit soort XAndYComponent aanmaken; ze zijn niet flexibel genoeg om changes aan slechts 1 deel van de implementaties op te vangen.

Reply:
hmm, dan expose je implementatie details van hoe die component gerenderd word naar buiten toe.
Het enige wat de gebruiker van de CollectionOrAssignmentBlocks om geeft zijn de props die exposed zijn. Hoe die onder de moterkap doorgegeven worden moet die gebruiker niet weten.

Idealieter zouden dit geen X en Y componenten zijn maar iets zoals: BlockEditList die enkel omgaat met BlockItemBase[]. 
Als die lijst van blokken in de toekomst anders gerenderd moet worden voor collecties dan voor assignments, moet die implementatie op dat punt opgesplitst worden. Momenteel is de implementatie nagenoeg identiek en willen we dus zoveel mogelijk code duplicatie vermijden.

Reply:
Omdat er geen verdere logica wordt uitgevoerd binnen `CollectionOrAssignmentBlocks` (bv. `permissions`-prop ipv `canPlay`-prop) is het niet aan te raden om properties van subcomponenten te dupliceren op hun implementerende wrappers. Om flexibiliteit tussen implementaties zonder duplicate typing te behouden doe je iets zoals;

https://github.com/viaacode/avo2-client/blob/epic/search-and-collect-assignments/src/assignment/hooks/assignment-content-modals.tsx#L16-L20
+
https://github.com/viaacode/avo2-client/blob/epic/search-and-collect-assignments/src/assignment/hooks/assignment-content-modals.tsx#L105-L106
+
https://github.com/viaacode/avo2-client/blob/epic/search-and-collect-assignments/src/assignment/views/AssignmentEdit.tsx#L204-L217
&
https://github.com/viaacode/avo2-client/blob/epic/search-and-collect-assignments/src/assignment/views/AssignmentCreate.tsx#L101-L105

Reply:
3 modals samen steken in 1 hook lijkt me ook redelijk vuil.
Bij collections edit hebben we gewoon de volledige edit en create in dezelfde component gebouwd: CollectionEdit
en dan kan je met een paar if's op collectionId, weten of het om een create of een update gaat. verder is 80% van de implementatie identiek.
Dan hoef je zelfs geen modals te exporteren naar een custom hook